packages/block-library/src/post-navigation-link/edit.js

```diff
@@ -47,6 +59,40 @@ export default function PostNavigationLinkEdit( {
 			[ `has-text-align-${ textAlign }` ]: textAlign,
 		} ),
 	} );
+
+	const taxonomies = useSelect(
+		( select ) => {
+			const { getTaxonomies } = select( coreStore );
+			const filteredTaxonomies = getTaxonomies( {
+				type: postType,
+				per_page: -1,
+				context: 'view',
+			} );
+			return filteredTaxonomies;
+		},
+		[ postType ]
+	);
+	const getTaxonomyOptions = () => {
+		const selectOption = {
+			label: __( 'Unfiltered' ),
+			value: '',
+		};
+		const taxonomyOptions = ( taxonomies ?? [] )
+			.filter(
+				( tax ) =>
+					tax.slug !== 'nav_menu' &&
+					tax.slug !== 'wp_pattern_category'
```

Comment:
I simply deleted the filter and nothing changed in the interface. I am not sure what changed since November but neither the nav menu, pattern categories or the admin views are showing up now. Custom taxonomies, for standard and for custom post types are showing correctly as well.

Reply:
I don't know about the navigation menu and admin view, but it seems that `public` has been changed to `false` in the pattern category.

- https://core.trac.wordpress.org/ticket/59569
- #55748